GRDB/Record/FetchableRecord.swift

```diff
@@ -614,3 +632,87 @@ public enum DatabaseDateDecodingStrategy {
+    private static func convertFromSnakeCase(_ stringKey: String) -> String {
+        guard !stringKey.isEmpty else { return stringKey }
+
+        var words: [Range<String.Index>] = []
+        // The general idea of this algorithm is to split words on transition from lower to upper case, then
+        // on transition of >1 upper case characters to lowercase
+        //
+        // myProperty -> my_property
+        // myURLProperty -> my_url_property
+        //
+        // We assume, per Swift naming conventions, that the first character of the key is lowercase.
+        var wordStart = stringKey.startIndex
+        var searchRange = stringKey.index(after: wordStart)..<stringKey.endIndex
+
+        // Find next uppercase character
+        while let upperCaseRange = stringKey.rangeOfCharacter(from: CharacterSet.uppercaseLetters,
+                                                              options: [], range: searchRange) {
+            let untilUpperCase = wordStart..<upperCaseRange.lowerBound
+            words.append(untilUpperCase)
+
+            // Find next lowercase character
+            searchRange = upperCaseRange.lowerBound..<searchRange.upperBound
+            guard let lowerCaseRange = stringKey.rangeOfCharacter(from: CharacterSet.lowercaseLetters,
+                                                                  options: [], range: searchRange) else {
+                // There are no more lower case letters. Just end here.
+                wordStart = searchRange.lowerBound
+                break
+            }
+
+            // Is the next lowercase letter more than 1 after the uppercase? If so, we encountered a group of
+            // uppercase letters that we should treat as its own word
+            let nextCharacterAfterCapital = stringKey.index(after: upperCaseRange.lowerBound)
+            if lowerCaseRange.lowerBound == nextCharacterAfterCapital {
+                // The next character after capital is a lower case character and therefore not a word boundary.
+                // Continue searching for the next upper case for the boundary.
+                wordStart = upperCaseRange.lowerBound
+            } else {
+                // There was a range of >1 capital letters. Turn those into a word, stopping at the capital
+                // before the lower case character.
+                let beforeLowerIndex = stringKey.index(before: lowerCaseRange.lowerBound)
+                words.append(upperCaseRange.lowerBound..<beforeLowerIndex)
+
+                // Next word starts at the capital before the lowercase we just found
+                wordStart = beforeLowerIndex
+            }
+            searchRange = lowerCaseRange.upperBound..<searchRange.upperBound
+        }
+        words.append(wordStart..<searchRange.upperBound)
+        let result = words.map({ (range) in
+            return stringKey[range].lowercased()
+        }).joined(separator: "_")
```

Comment:
Not what I would expect for `convertFromSnakeCase` 😉 Do tests catch this?

Reply:
It's mostly a naming issue. Let's say we're _decoding_ this `FetchableRecord`:

```swift
struct Record: FetchableRecord {
	static let databaseKeyDecodingStrategy = DatabaseKeyDecodingStrategy.convertFromSnakeCase
	var recordID: String
}
```

During the decoding process, the decoder gets the key as `recordID`, which we first need to convert _to_ snake case to look it up in the `Row` (which uses the SQL conventional `record_id`). It seems that `convertToSnakeCase` would be a better name then, but from a public API perspective that's not true:

```
struct Record: FetchableRecord {
	static let databaseKeyDecodingStrategy = DatabaseKeyDecodingStrategy.convertToSnakeCase
	var recordID: String
}
```

Now it reads as if `Record` will have its keys converted _to_ snake case, which is confusing as `recordID` is using camel case.

My suggestion is to make the public API as simple as possible, and possibly change some names around internally, but either way it's not super intuitive. Let me know what you prefer and I can make that change!